### PR TITLE
Con 1165 netsparker wildcard

### PR DESCRIPTION
**Pull request by annapoltieva**
**JIRA** [CON-1165](https://kennasecurity.atlassian.net/browse/CON-1165)

**Problem**
Lowes has a large number of scheduled ids and it is hard for them to find them.

**Solution**
Turn netsparker_scheduled_ids argument into an optional one and pull all the scheduled ids from the API if this argument is not given or set to *

### CODE REVIEW

**Review by mukhansk — APPROVED** [review]

File: tasks/connectors/netsparker/lib/netsparker_client.rb [hunk 1]
```diff
@@ -23,6 +23,23 @@ def get_last_scan_vulnerabilities(schedule_id)
...
+          schedule_ids = []
+          loop do
+            response = http_get(list_scheduled_url(page), @headers)
+            scheduled_scan_result = JSON.parse(response)
+            schedule_ids.push(*scheduled_scan_result.fetch("List").map { |scan| scan.fetch("Id") }.uniq)
+            break if scheduled_scan_result["IsLastPage"]
+
+            page += 1
+          end
+          schedule_ids.uniq
```
Comment (jscalvo) [inline tasks/connectors/netsparker/lib/netsparker_client.rb:37]: Why is `uniq` called again? Is called in the loop also.

File: tasks/connectors/netsparker/lib/netsparker_client.rb [hunk 2]
```diff
@@ -23,6 +23,23 @@ def get_last_scan_vulnerabilities(schedule_id)
...
+        rescue KeyError
+          print_error "There are no scheduled scans"
+          exit 1
```
Comment (jscalvo) [inline tasks/connectors/netsparker/lib/netsparker_client.rb:40]: There is a helper for above 2 lines:
```ruby
fail_task "There are no scheduled scans"
```

File: tasks/connectors/netsparker/lib/netsparker_client.rb [hunk 3]
```diff
@@ -23,6 +23,23 @@ def get_last_scan_vulnerabilities(schedule_id)
           JSON.parse(response)
         end
 
+        def receive_all_schedule_ids
```
Comment (jscalvo) [inline tasks/connectors/netsparker/lib/netsparker_client.rb:26]: I'd change the method name to something like `get_all_schedule_ids` or `retrieve_all_scheduled_ids`.

**Review by jscalvo — CHANGES_REQUESTED** [review]

File: tasks/connectors/netsparker/lib/netsparker_client.rb (hunk 1 above)
Comment (annapoltieva) [reply tasks/connectors/netsparker/lib/netsparker_client.rb:37]: My bad. It should just be called once outside of the loop

**Review by jscalvo — APPROVED** [review]

**Pull request merged**